internal/nanny/assigner/assigner.go

```diff
@@ -181,62 +176,54 @@ func (a *Assigner) registerApp(ctx context.Context, app string) error {
 	return nil
 }
 
-// registerGroup registers the provided group in the appropriate app's
-// state.
-func (a *Assigner) registerGroup(ctx context.Context, gid *GroupId) error {
-	_, _, err := a.applyAppState(ctx, gid.App, func(state *AppState) bool {
-		for _, existing := range state.GroupIds {
-			if proto.Equal(gid, existing) {
+// registerReplicaSet registers the provided Kubernetes ReplicaSet in the
+// appropriate app's state.
+func (a *Assigner) registerReplicaSet(ctx context.Context, rid *ReplicaSetId) error {
+	_, _, err := a.applyAppState(ctx, rid.App, func(state *AppState) bool {
+		for _, existing := range state.ReplicaSetIds {
+			if proto.Equal(rid, existing) {
 				return false
 			}
 		}
-		state.GroupIds = append(state.GroupIds, gid)
+		state.ReplicaSetIds = append(state.ReplicaSetIds, rid)
 		return true
 	})
 	if err != nil {
-		return fmt.Errorf("register group %v: %v", gid, err)
+		return fmt.Errorf("register replica set %v: %v", rid, err)
 	}
 	return nil
 }
 
-func (a *Assigner) RegisterComponentToStart(ctx context.Context, req *protos.ComponentToStart) error {
-	id, err := uuid.Parse(req.DeploymentId)
+// RegisterActiveComponent registers the given component has been activated. It
+// returns the name of the Kubernetes ReplicaSet that should host the component.
+func (a *Assigner) RegisterActiveComponent(ctx context.Context, req *nanny.ActivateComponentRequest) error {
+	id, err := uuid.Parse(req.Config.Deployment.Id)
```

Comment:
In a future change, we can avoid parsing the uuid and update the `store.` methods?

Reply:
Thanks. Added a TODO. WIl do so in subsequent change.